### PR TITLE
HSEARCH-1951 Validate sort type used for query

### PR DESCRIPTION
**Pull request by DavideD**
https://hibernate.atlassian.net/browse/HSEARCH-1951

### CODE REVIEW

File: engine/src/test/java/org/hibernate/search/test/sorting/SortingTest.java [hunk 1]
```diff
@@ -46,8 +48,13 @@
  */
 public class SortingTest {
 
+	private static final String SORT_TYPE_ERROR_CODE = "HSEARCH000306";
```
Comment (gsmet) [inline engine/src/test/java/org/hibernate/search/test/sorting/SortingTest.java]: Should be 307. Probably a merge problem.

File: engine/src/main/java/org/hibernate/search/query/engine/impl/AbstractHSQuery.java [hunk 2]
```diff
@@ -231,31 +240,185 @@ public ExtendedSearchIntegrator getExtendedSearchIntegrator() {
...
+			throw LOG.distanceSortRequiresSpatialField( sortField.getField() );
+		}
+	}
+
+	private FieldDescriptor findFieldDescriptor(ExtendedSearchIntegrator extendedIntegrator, Iterable<Class<?>> targetedEntities, String field) {
+		if ( field == null ) {
+			return null;
+		}
+		for ( Class<?> clazz : targetedEntities ) {
+			IndexedTypeDescriptor indexedTypeDescriptor = extendedIntegrator.getIndexedTypeDescriptor( clazz );
```
Comment (gsmet) [inline engine/src/main/java/org/hibernate/search/query/engine/impl/AbstractHSQuery.java]: Is there a reason why you used the FieldDescriptor API? From what I know, it's not recommended to use it in our code. See for instance Gunnar's comment in https://hibernate.atlassian.net/browse/HSEARCH-2070 .

File: engine/src/test/java/org/hibernate/search/test/sorting/SortingTest.java (hunk 1 above)
Comment (DavideD) [inline engine/src/test/java/org/hibernate/search/test/sorting/SortingTest.java]: Ops, I didn't include the change in the PR

File: engine/src/main/java/org/hibernate/search/query/engine/impl/AbstractHSQuery.java (hunk 2 above)
Comment (DavideD) [inline engine/src/main/java/org/hibernate/search/query/engine/impl/AbstractHSQuery.java]: This part was already like this, I just refactored the code.
I'll check if it works using the DocumentFieldMetadata

**Comment by DavideD** [comment]
@gsmet Thanks for the feedback. I've push-forced some changes. Now I'm using the DocumentFieldMetadata for spatial as well..

**Comment by Sanne** [comment]
Looks great. I'm backporting it .. not trivial as:
 - the patched files didn't exist
 - some of the used interfaces didn't exist either

**Comment by Sanne** [comment]
I've merged it but made one little change: 6d390a0

I realized the need for this after backporting.

**Pull request merged**